### DISCUSSION TITLE
How to use a nutmeg equation to set a parameter?

### DISCUSSION THREAD

**Discussion by Swontu** [ra3xdh/qucs_s#657] [Q&A] [answered]
Hi there,

can someone explain to me what I'm doing wrong in this trivial toy example, when using an Nutmeg equation to set a parameter?
![image](https://github.com/ra3xdh/qucs_s/assets/164629394/fc772796-3fdd-472b-b20b-169d93d8e1a2)
![image](https://github.com/ra3xdh/qucs_s/assets/164629394/27e08510-b9b8-4c74-b57b-0d4ba90a696e)

Interestingly, if I change the simulator to SpiceOus and insert an equation, qucs-s inserts a regular equation (not Nutmeg). Now when I change back to the ngspice-simulator, the simulation runs without an error and the plot shows the expected result:
![image](https://github.com/ra3xdh/qucs_s/assets/164629394/b4fc57b5-9e0e-4391-9ebc-2b4f6d5d7e5b)
The netlist now also includes a .PARAM statement, which was missing before:
![image](https://github.com/ra3xdh/qucs_s/assets/164629394/ecd02a6f-4088-4403-aeef-830f93c200df)

My setup:
Windows 10
ngspice-42
Qucs-S Portable 24.1.0

Thanks in advance for any help!

**Comment by ra3xdh** [comment, marked as answer]
> can someone explain to me what I'm doing wrong in this trivial toy example, when using an Nutmeg equation to set a parameter?

Nutmeg is SPICE postprocessor. These equations are evaluated *before* simulation. You cannot substitute parameters using Nutneg equations. Use `.PARAM` instead. This device could be found in *equations* group.

> qucs-s inserts a regular equation (not Nutmeg).

It is qucsator equation. It supports only limited set of mathematical functions that are compatible between all simulators. I am hiding it in SPICE mode to prevent its insertion by accident.
